lib/server/generate.js

```diff
@@ -180,7 +180,7 @@ function execute() {
         {rawContent}
       </DocsLayout>
     );
-    const str = renderToStaticMarkup(docComp);
+    const str = '<!DOCTYPE html>' + renderToStaticMarkup(docComp);
```

Comment:
Is there a reason why you could not just put this in the render section of the site in `Site.js`, so you only have to do it one time?

Right above here, before the `<html>` tag? https://github.com/facebook/Docusaurus/blob/master/lib/core/Site.js#L56

Reply:
It would be great yes, but React seems to not allow us do to this and raised an error about Doctype, here is an issue with this point on React repo https://github.com/facebook/react/issues/1035.

Reply:
@Tonours Ugh. That's unfortunate. So it might be that your way is the only/best way to do it - injecting `<DOCTYPE html>` inline with any call to `renderToStaticMarkup`. 

@rickyvetter would you know a better way to do this given what we are seeing in  facebook/react#1035

Reply:
This is quite a verbose way of injecting it even though the issue suggests that string manipulation *is* the only way of doing this. How about wrapping `renderToStaticMarkup` to keep it DRY?

```js
const renderToStaticMarkup = require('react-dom/server').renderToStaticMarkup;
...
// Some better name (the hard things in coding...):
const fullyRenderToStaticMarkup = (...args) => `<!DOCTYPE html>${renderToStaticMarkup(...args)}`;
```
_NB: Could replace destructuring (...args) with explicit arguments if not supported or  type inference is needed at some point_